src/onConsent.ts

```diff
@@ -0,0 +1,23 @@
+import { onConsentChange } from './onConsentChange';
+import type { ConsentState } from './types';
+
+/**
+ * A promise wrapper around `onConsentChange` that resolves the initial consent state
+ *
+ * This will only resolve once whereas callbacks passed to `onConsentChange`
+ * are executed each time consent state changes. Avoid using this function
+ * in contexts where subsequent consent states must be listened for.
+ *
+ * @returns Promise<ConsentState>
+ */
+const onConsent = (): Promise<ConsentState> =>
+	new Promise<ConsentState>((resolve, reject) => {
+		onConsentChange((consentState) => {
```

Comment:
> which adds a callback to the callBackQueue, and never removes it.

> So it'll callback on future consents regardless?

Yes you're right the callback will still exist but because the promise is already resolved nothing will happen. I see this as the equivalent of a callback that decides to do nothing with the consent state on a subsequent call.

From a resource point of a view it is a little wasteful in the event of consent changing post page load but we don't (afaik) currently have a method to unregister a callback.

Slight tangent - I think we've talked before moving to a model where we reload the page if consent changes. I think that would remove a lot of the complexity in dynamically responding to changing consent. Perhaps that is something we could move forwards on?

Reply:
> Perhaps that is something we could move forwards on?

It's [ticketed](https://trello.com/c/8fvDoDhM/609-update-onconsentchange-in-the-cmp-wrapper-refresh-the-page-if-consent-changes) but unlikely to happen soon.